doc/release-notes: initial cleanup for 24.11

getchoo opened the pull request:
Most of this is meant to improve grammar and word flow, correct errors, and be more consistent in how certain points are described. It should help serve as a basis for the rest of our work up until 24.11's final release

CC @GetPsyched as the other editor for this release

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

getchoo committed:
doc/release-notes: improve consistency and grammar in new modules section

getchoo committed:
doc/release-notes: use more concise descriptions in new modules section

getchoo committed:
doc/release-notes: remove MenhirLib from new modules section

This is not a module

getchoo committed:
doc/release-notes: cleanup highlights

The main goal is to make these highlights a bit more concise,
consistently link to newly introduced option definitions, and (somewhat
subjectively) improve word choice to avoid repetition and have a better
flow

getchoo committed:
doc/release-notes: cleanup backward incompatibilities section

The main goal is to make these points a bit more concise, have better
consistency in how removals and deprecations are described, and improve
grammar/flow

getchoo committed:
doc/release-notes: cleanup notable changes section

The main goal is to make these points a bit more concise, fix errors,
and (somewhat subjectively) improve word choice to avoid repetition and
have a better flow

getchoo committed:
doc/release-notes: correctly place backwards incompatible changes

These changes seemed to have been placed in the wrong section

getchoo committed:
doc/release-notes: use title case for migrations section

jopejoe1 commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -188,40 +182,39 @@
-  NixOS does not officially support this configuration and might cause your system to be unbootable in future versions. You are on your own.
+- `systemd.enableUnifiedCgroupHierarchy` has been removed.
+  In systemd 256, support for cgroup v1 ('legacy' and 'hybrid' hierarchies) is now considered obsolete and systemd will refuse to boot under it by default.
+  To forcibly re-enable cgroup v1 support, you can `set boot.kernelParams = [ "systemd.unified_cgroup_hierarchy=0" "SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1" ]`.
```
The `set` should probly be before the formated string and not in it.

```suggestion
  To forcibly re-enable cgroup v1 support, you can set `boot.kernelParams = [ "systemd.unified_cgroup_hierarchy=0" "SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1" ]`.
```

GetPsyched reviewed (approved):
That was a really thorough cleanup, thanks! Just one minor non-blocking nit

GetPsyched commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -385,23 +388,23 @@
   with *runner authentication tokens* until the release of GitLab 18.0.
 
-- `gitlab` has been updated from 16.x to 17.x and requires at least `postgresql` 14.9, as stated in the [documentation](https://docs.gitlab.com/17.1/ee/install/requirements.html#postgresql-requirements). Check the [upgrade guide](#module-services-postgres-upgrading) in the NixOS manual on how to upgrade your PostgreSQL installation.
+- `gitlab` has been updated from 16.x to 17.x and requires at `postgresql` >= 14.9, as stated in the [documentation](https://docs.gitlab.com/17.1/ee/install/requirements.html#postgresql-requirements). Check the [upgrade guide](#module-services-postgres-upgrading) in the NixOS manual on how to upgrade your PostgreSQL installation.
```
```suggestion
- `gitlab` has been updated from 16.x to 17.x and requires `postgresql` >= 14.9, as stated in the [documentation](https://docs.gitlab.com/17.1/ee/install/requirements.html#postgresql-requirements). Check the [upgrade guide](#module-services-postgres-upgrading) in the NixOS manual on how to upgrade your PostgreSQL installation.
```

GetPsyched commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -452,63 +457,66 @@
   The systemd service now uses `DynamicUser`/`StateDirectory` and the `user` and `group` options have been deprecated.
 
-- The `openlens` package got removed, suggested replacement `lens-desktop`
+- `openlens` was removed. It is recommended to use `lens-desktop` instead
```
```suggestion
- `openlens` was removed. It is recommended to use `lens-desktop` instead.
```

GetPsyched reviewed (commented):
Minor nits + conflicts now

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -7,51 +7,51 @@
   Starting with release 25.05, the minimum supported version will be macOS Big Sur 11, and we cannot guarantee that packages will continue to work on older versions of macOS.
   Users on old macOS versions should consider upgrading to a supported version (potentially using [OpenCore Legacy Patcher](https://dortania.github.io/OpenCore-Legacy-Patcher/) for old hardware) or installing NixOS.
-  If neither of those options are viable and you require new versions of software, [MacPorts](https://www.macports.org/) supports back to Mac OS X Snow Leopard 10.6.
+  If neither of those options are viable and you require new versions of software, [MacPorts](https://www.macports.org/) supports versions back to Mac OS X 10.6 Snow Leopard.
```
FWIW this was intentional for consistency with the other releases mentioned in the note and with how Apple usually describes contemporary OS releases, though they’ve been pretty inconsistent historically.

getchoo replied:
OS X to my knowledge has always followed this sort of version scheming (see the [wikipedia article](https://en.wikipedia.org/wiki/Mac_OS_X_Snow_Leopard) for example), so I figured it would be better to stay consistent with that. Not really sure how to refer to macOS releases though, since yeah like you said it's inconsistent; it's probably best to just leave those

emilazy replied:
So Apple almost universally used “v10.6” rather than “10.6” back then from what I’ve seen, but the ordering of the names and version varied a bit, albeit with the name‐first form being predominant in the relatively rare cases where Apple used both. You can see “Mac OS X Snow Leopard v10.6” in the press releases [announcing its release](https://www.apple.com/newsroom/2009/06/08Apple-Unveils-Mac-OS-X-Snow-Leopard/) and [the release of Safari 4](https://www.apple.com/newsroom/2009/06/08Apple-Releases-Safari-4-The-Worlds-Fastest-Most-Innovative-Browser/), but “Mac OS X v10.6 Snow Leopard” on the [technical specifications page](https://support.apple.com/en-gb/112591). The link in the footer of the Apple site from back then was “10.6 Snow Leopard”, and the [contemporary version of the technical specs page](https://web.archive.org/web/20090714080723/http://www.apple.com/macosx/specs.html) also talked about “Mac OS X v10.6 Snow Leopard”. The [Mac OS X Lion installer](https://support.apple.com/en-gb/106383) and [Mac App Store update](https://support.apple.com/en-gb/106470) system requirements list “OS X Snow Leopard 10.6” (anachronistically without the “Mac”, and the Lion installer inconsistently lists the version first in the text description – so perhaps this is due to some later format normalization). This list of [security updates from 2011 to 2012](https://support.apple.com/en-gb/101444) talks about “OS X Lion v10.7.2” and so on, though not for Snow Leopard itself.

On the other hand, for contemporary versions, the “macOS Sequoia 15” format is seemingly almost universally used when Apple (rarely) use both the name and version, e.g. [in the user guide](https://support.apple.com/en-gb/guide/mac-help/apd07d671600/mac), the [SDK release notes](https://developer.apple.com/documentation/macos-release-notes/macos-15-release-notes), and the [security release overview](https://support.apple.com/en-us/100100).

In general Apple didn’t pair the code names and releases as often back then, so it’s harder to point to a canonical format. I went for consistency with the versions people actually care about and Apple’s current usage here. Anyway all of this is of course completely irrelevant and you can do whatever here, but I lost some 10 minutes researching this when I was touching up the wording of the message in the first place so I figured I can at least record my findings for posterity in the unlikely case that this ever comes up again in the future.

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -452,63 +457,66 @@
+- `mikutter` has been removed, as the package was broken and had no maintainers in nixpkgs.
 
 - `ffmpeg_5` has been removed. Please use the unversioned `ffmpeg`,
-  pin a newer version, or if necessary pin `ffmpeg_4` for compatibility.
+  pin a newer version, or pin `ffmpeg_4` for compatibility if necessary.
```
This release note was already rewritten on the current `staging-next` (and will change again in https://github.com/NixOS/nixpkgs/pull/346552).

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -452,63 +457,66 @@
-  removed as nobody was willing to maintain them and they were stuck on
-  an unmaintained version that required FFmpeg 4; please see [pull
+- `tvheadend` package and the `services.tvheadend` module have been
+  removed due to lack of maintenance in nixpkgs and being stuck on
```
```suggestion
  removed due to lack of maintenance in Nixpkgs and being stuck on
```

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -452,63 +457,66 @@
-  an unmaintained version that required FFmpeg 4; please see [pull
+- `tvheadend` package and the `services.tvheadend` module have been
+  removed due to lack of maintenance in nixpkgs and being stuck on
+  a version that required FFmpeg 4. Please see the related [pull
```
The version that requires FFmpeg 4 is from a release branch that is unmaintained upstream, so I think removing “unmaintained” here loses some nuance.

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -572,41 +580,34 @@
-- Mikutter was removed because the package was broken and had no maintainers.
-
-- Gollum was upgraded to major version 6. Read their [migration notes](https://github.com/gollum/gollum/wiki/6.0-Release-Notes).
+- `security.pam.u2f` now follows RFC42; all module options are now configurable through `security.pam.u2f.settings`.
```
It might be good to link [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md).

getchoo replied:
I'm actually not really sure how we should go about these since there's a few mentions of it. IMO any mention of RFC42 could be removed in favor of just referring to them as freeform modules, since users will probably be more familiar with that concept (given RFCs are more of a developer oriented thing)

Would like some feedback on this, though

emilazy commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -572,41 +580,34 @@
+- `restic` module now has an option for inhibiting system sleep while backups are running, defaulting to off (not inhibiting sleep). Available as [`services.restic.backups.<name>.inhibitsSleep`](#opt-services.restic.backups._name_.inhibitsSleep).
 
-- The arguments from [](#opt-services.postgresql.initdbArgs) now get shell-escaped.
+- The arguments from [services.postgresql.initdbArgs](#opt-services.postgresql.initdbArgs) now get shell-escaped.
```
Doesn’t the manual generator already handle including the name here? If not, shouldn’t this at least be in backticks?

getchoo replied:
It does! TIL

GetPsyched commented:
What's the status of this PR? Let's get this merged soon.

getchoo committed:
doc/release-notes: RFC 42 -> freeform options

"Freeform options" is a term that's a bit more meaningful to regular
users who don't follow development, and thus does a better job at
describing the changes

getchoo commented:
Resolved merge conflicts

GetPsyched commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -138,63 +135,59 @@
+- [Veilid](https://veilid.com), a privacy-focused, headless server for data sharing and messaging on a peer-to-peer network. Available as [services.veilid](#opt-services.veilid.enable).
 
-- [Fedimint](https://github.com/fedimint/fedimint), a module based system for building federated applications (Federated E-Cash Mint). Available as [services.fedimintd](#opt-services.fedimintd).
+- [Fedimint](https://github.com/fedimint/fedimint), a module based system for building federated applications (Federated E-Cash Mint). Available as [services.fedimintd](#opt-services.fedimintd.enable).
```
> bad local reference, id #opt-services.fedimintd.enable not known

https://github.com/NixOS/nixpkgs/actions/runs/11451771665/job/31861545904?pr=346059#step:5:137

Getting this error because [the appendix page for fedimintd](https://nixos.org/manual/nixos/unstable/options#opt-services.fedimintd) doesn't have the `.enable` bit.

A few loosely related notes:
1. Is there no convention for this? Having or not having `.enable` (AFAIK that appendix is auto generated, so it should have some consistency)
2. IIRC the hyperlinks for module options are auto-filled by nixos-render-docs, so we probably don't need to specify that anywhere? Would need confirmation on this.

getchoo replied:
> Having or not having .enable

As long as the `.enable` option exists, yes. As it turns out, fedimintd just [doesn't have one](https://github.com/NixOS/nixpkgs/blob/729d9ddec137ab399cfdab567072896de8a0ccf8/nixos/modules/services/networking/fedimintd.nix#L183-L187)

> IIRC the hyperlinks for module options are auto-filled by nixos-render-docs

As in `[](#opt-services.foo.enable)`? That does work, but it would end up making the hyperlink the literal `services.foo.enable` rather than only `services.foo` -- and most modules won't have a single root `services.foo` option

getchoo commented:
Rebased to get rid of merge conflicts.

GetPsyched commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -144,65 +141,61 @@
-
 - [tiny-dfr](https://github.com/WhatAmISupposedToPutHere/tiny-dfr), a dynamic function row daemon for the Touch Bar found on some Apple laptops. Available as [hardware.apple.touchBar.enable](options.html#opt-hardware.apple.touchBar.enable).
 
+- [Zapret](https://github.com/bol-van/zapret), a DPI bypass tool. Available as [services.zapret](option.html#opt-services.zapret.enable).
```
Any reason this was moved below `tiny-dfr`? (not blocking on this, it's perfectly fine -- just curious)

getchoo replied:
Nope, just a result of the rebase :p

GetPsyched reviewed (approved):
LGTM!

getchoo commented:
Rebased to get rid of merge conflicts.

GetPsyched reviewed (approved):
Looked through rebase diff, all good
